src/core.ts

```diff
@@ -112,13 +118,13 @@ export default class AgentCore extends Store {
     }
   }
 
-  _updateInteractCache (data: Object): void {
+  _updateInteractCache (data: Interact): void {
```

Comment:
You did not have to change this part. Because it have validator

Reply:
validator function argument needs typing causes `noImplicitAny` compile. 
I guess that the arg type should be 'Interact' not `Object`,  and `_updateInteractCache` arg 
 type should be same.